server.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const http = require('http');
+const url = require('url');
+const querystring = require('querystring');
+const cowsay = require('cowsay');
+const parseBody = require('./lib/parse-body.js');
+const PORT = process.env.PORT || 3000;
+
+const server = http.createServer(function(request, response) {
+  request.url = url.parse(request.url);
+  request.url.query = querystring.parse(request.url.query);
+
+  // console.log('req url:', request.url); //returns url object
+  // console.log('req querystring:', request.url.query); //returns empty object
+  // console.log(request.method); //returns GET method
+
+  if(request.method === 'POST') {
+    parseBody(request, function(err) {
+      if (err) console.log(err);
+      console.log('POST request body:', request.body);
+      if(request.url.pathname === '/cowsay' && request.body.text !== undefined) {
+        response.writeHead(200, {'Content-Type': 'text/plain'});
```

Comment:
Another nitpick, I find it more convenient to use req and res over the simple premise that I don't have to type out the full words.  

Also, instead of request.body.text !== undefined you could just check for request.body.text